druid/src/widget/mod.rs

```diff
@@ -127,6 +129,22 @@ pub trait Widget<T> {
     /// [`Command`]: struct.Command.html
     fn event(&mut self, ctx: &mut EventCtx, event: &Event, data: &mut T, env: &Env);
 
+    /// Handle a life cycle notification.
+    ///
+    /// This method is called to notify your widget of certain special events,
+    /// (available in the [`LifeCycle`] enum) that are generally related to
+    /// changes in the widget graph or in the state of your specific widget.
+    ///
+    /// In general, a widget is not expected to mutate the application state
```

Comment:
Why the "in general" qualification? This seems clear from the signature; the only way it can do that is internal mutability, and I think there's a pretty clear convention against that. Maybe I'm being overly picky though...